src/Product.php

```diff
@@ -626,6 +629,26 @@ public function processImages($data)
         return $this;
     }
 
+    /**
+     * @return array|null
+     */
+    protected function getAvailableImage(array $data)
+    {
+        foreach ($data as $image) {
+            if (empty($imageAvailabilities = $image['availabilities'])) {
+                continue;
+            }
+
+            foreach ($imageAvailabilities as $imageAvailability) {
+                if ($imageAvailability['type'] === self::AVAILABILITY_SHOP) {
+                    return $image;
+                }
+            }
+        }
+
+        return null;
```

Comment:
We could also just return an empty array here, meaning that no images were found that are assigned to the current store.